statsmodels/tsa/vector_ar/var_model.py

```diff
@@ -261,85 +280,263 @@ def var_loglike(resid, omega, nobs):
-        super(VAR, self).__init__(endog, None, dates, freq, missing=missing)
+    Y_array = np.asarray(Y)
+    if data_util._is_structured_ndarray(Y_array):
+        Y_array = Y_array.view((float, len(Y_array.dtype)))
+
+    if maxlags is None:
+        maxlags = int(round(12*(len(Y)/100.)**(1/4.)))
+
+    k_trend = util.get_trendorder(trend)
+    ics = defaultdict(list)
+    for k_ar in range(maxlags + 1):
+        # exclude some periods so same nobs used for each lag order
+        res = _estimate_var_ic(Y_array, k_ar, k_trend, trim=maxlags - k_ar,
+                               trend=trend)
+        ics["aic"].append(res[0])
+        ics["bic"].append(res[1])
+        ics["hqic"].append(res[2])
+        ics["fpe"].append(res[3])
+
+    selected_orders = dict((k, mat(v).argmin())
+                           for k, v in ics.iteritems())
+    try:
+        nlags = selected_orders[ic]
+    except:
+        KeyError("ic %s is not understood" % ic)
+    return VAR(Y, nlags).fit(trend=trend)
+
+class VAR(tsbase.TimeSeriesModel):
+    __doc__ = tsbase._tsa_doc % {"model" : _var_model_doc,
+                                 "params" : _var_params_doc,
+                                 "extra_params" : tsbase._missing_param_doc,
+                                 "extra_sections" : _var_reference_doc}
+    def __init__(self, Y, nlags=None, dates=None, names=None, freq=None,
+                 missing='none'):
+        if nlags is None:
+            from warnings import warn
+            warn("In the 0.6.0 release nlags will not be optional in the "
+                 "model constructor.", FutureWarning)
+        else:
+            self.k_ar = nlags
+        super(VAR, self).__init__(Y, None, dates, freq, missing=missing)
         if self.endog.ndim == 1:
-            raise ValueError("Only gave one variable to VAR")
-        self.y = self.endog #keep alias for now
-        self.neqs = self.endog.shape[1]
+            raise ValueError("Only one variable given to VAR. Expects > 1.")
+        if names is not None:
+            import warnings
+            warnings.warn("The names argument is deprecated and will be "
+                          "removed in 0.6.0. Use a pandas object or structured "
+                          "array for names.", FutureWarning)
+            self.names = names
+        else:
+            self.names = self.endog_names
+        self.Y = self.endog #keep alias for now
+        self.neqs = self.Y.shape[1]
+
 
     def _get_predict_start(self, start, k_ar):
         if start is None:
             start = k_ar
         return super(VAR, self)._get_predict_start(start)
 
+    def loglike(self, params, k_ar=None):
+        r"""
+        Returns the value of the VAR(p) log-likelihood.
+
+        Parameters
+        ----------
+        params : ndarray
+            The parameters in the order given in VARResults.
+        k_ar : int, optional
+            The number of lags if the equation has not yet been fit.
```

Comment:
`if the equation has not yet been fit` 
problem because this is not initialized in `__init__`